packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -2108,3 +2108,95 @@ class _AnimatedPhysicalModelState extends AnimatedWidgetBaseState<AnimatedPhysic
     );
   }
 }
+
+/// Animated version of [FractionallySizedBox] which automatically transitions the
+/// child's size over a given duration whenever the given [widthFactor] or
+/// [heightFactor] changes, as well as the position whenever the given [alignment]
+/// changes.
+///
+/// For the animation, you can choose a [curve] as well as a [duration] and the
+/// widget will automatically animate to the new target [widthFactor] or
+/// [heightFactor].
```

Comment:
There's also a video in some of the other related widgets, like
```dart
/// Here's an illustration of what this can look like, using a [curve] of
/// [Curves.fastOutSlowIn].
/// {@animation 250 266 https://flutter.github.io/assets-for-api-docs/assets/widgets/animated_align.mp4}
```

I wasn't sure how to add those though

Reply:
Those are added in the [flutter/assets-for-api-docs](https://github.com/flutter/assets-for-api-docs/blob/addf8d091bd5b39e347a94f9d9257dcf614df47b/packages/diagrams/lib/src/implicit_animations.dart#L103) repo, like the one I linked to for `AnimatedContainer`.